src/world.rs

```diff
@@ -338,15 +334,15 @@ impl World {
 /// An opaque entity identifier.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Entity {
-    id: u32,
-    gen: NonZeroU32,
+    pub(crate) id: u32,
+    pub(crate) gen: NonZeroU32,
 }
 
 #[derive(Clone, Copy)]
-struct EntityMetadata {
-    gen: NonZeroU32,
-    ty: u32,
-    idx: u32,
+pub struct EntityMetadata {
+    pub gen: NonZeroU32,
+    pub ty: u32,
+    pub idx: u32,
```

Comment:
I think `pub(crate)` would suffice here, for the struct as well as the fields. Or is there a reason one could want to use this outside the crate?

Reply:
This is not public outside of the crate as it is not exported in `src/lib.rs`. `pub` is generally preferable to `pub(crate)` because it is a local decision where the next module up the hierarchy decides what is visible and what is not. `pub(crate)` is meant specifically for those cases where the type is basically `pub` but we still need to restrict some parts to the crate internals, e.g. `pub struct Entity` but `pub(crate) id: u32`.